### PR TITLE
Update dependencies in net-mail/offlineimap3

### PR DESCRIPTION
**Pull request by r7l**
The OfflineImap repo had [a PR merged recently](https://github.com/OfflineIMAP/offlineimap3/pull/102), to add support for keyring. This leads to a missing dependency when using your Ebuild.

I've added the missing dependency and also updated and restructured the dependeny handling in general. The original ebuild wasn't working when being used in [a `ROOT` variable environment](https://devmanual.gentoo.org/ebuild-writing/variables/index.html#root). The dependencies are not simply RDEPEND as they need to be present in the host system during install.

### CODE REVIEW

**Comment by ferki** [comment]
Thanks for the PR and for your patience, @r7l!

I found the overall proposal hard to follow and dissect because of the single commit containing several changes. The diff also contains additional EAPI and PYTHON_COMPAT changes which were not mentioned in the description, making me feel uncertain whether those were intentional or not.

In any case, I looked at the individual pieces I could identify here and upstream, and made several updates around the offlineimap3 ebuilds now:

- update PYTHON_COMPAT
- update EAPI
- add missing `keyring` dependency to live ebuild
- patch 8.0.0 to support python 3.11 properly
- fix a QA warning
- drop contrib-only dependencies
- fix BDEPENDs

Please let me know if it fixes your use case, and if not, please open an issue.

**Comment by r7l** [comment]
@ferki I've not had time to test your Ebuild in recent days. But i've just did and since you've made the RDEPEND dependencies just BDEPEND now, it still not usable in a situation when using ROOT variable. I'll end up with missing dependencies as they are all just used when building while they are actually both: Build dependencies and run dependencies. At least some of them. It builds fine but won't work when using it on in the ROOT environment:

```
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/offlineimap/utils/distro_utils.py", line 11, in <module>
    from platform import linux_distribution
ImportError: cannot import name 'linux_distribution' from 'platform' (/usr/lib/python3.11/platform.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.11/offlineimap", line 19, in <module>
    from offlineimap import OfflineImap
  File "/usr/lib/python3.11/site-packages/offlineimap/__init__.py", line 20, in <module>
    from offlineimap.init import OfflineImap
  File "/usr/lib/python3.11/site-packages/offlineimap/init.py", line 29, in <module>
    from offlineimap.utils.distro_utils import get_os_name
  File "/usr/lib/python3.11/site-packages/offlineimap/utils/distro_utils.py", line 13, in <module>
    from distro import linux_distribution
ModuleNotFoundError: No module named 'distro'
```

**Comment by ferki** [comment]
@r7l: thanks for the feedback!

I agree I might have made it even worse than before. I pushed some changes for this, though I still feel confused exactly which deps your use case would require in exactly which *DEPEND variables.

If you still have troubles, let's discuss it elsewhere (IRC, new issue, [Open Source Office Hours](https://cal.com/ferki/opensource), etc.)

It may be useful for me to see the exact code on its own about what you'd like to change for your use case (without other changes intermixed.)

There may be a new offlineimap3 release coming soon (:crossed_fingers:), and I'd love to get the ebuild sorted for that update if there's still bugs left.
